learning_resources_search/serializers.py

```diff
@@ -451,8 +459,58 @@ def get_previous(self, instance) -> str | None:
     def get_count(self, instance) -> int:
         return instance.get("hits", {}).get("total", {}).get("value")
 
+    def update_list_parents(self, hits):
+        """Fill in user_list_parents and learning_path_parents for users"""
+
+        request = self.context.get("request")
+        if request and request.user and request.user.is_authenticated:
+            learning_path_parents_dict = {}
+            user_list_parents_dict = {}
+            hit_ids = [hit.get("_id") for hit in hits]
+
+            # Get user_list_parents for all returned resources
+            user_list_parents = UserListRelationship.objects.filter(
+                parent__author=request.user, child_id__in=hit_ids
+            ).values("id", "child_id", "parent_id")
+            for parent in user_list_parents:
+                user_list_parents_dict.setdefault(str(parent["child_id"]), []).append(
+                    parent
+                )
+
+            if (
+                request.user.is_staff
+                or request.user.is_superuser
+                or request.user.groups.filter(name=GROUP_STAFF_LISTS_EDITORS).first()
+                is not None
+            ):
+                # Get learning path parents for all returned resources
+                learning_path_parents = LearningResourceRelationship.objects.filter(
+                    child__id__in=hit_ids,
+                    relation_type=LearningResourceRelationTypes.LEARNING_PATH_ITEMS.value,
+                ).values("id", "child_id", "parent_id")
+                for parent in learning_path_parents:
```

Comment:
this method is a bit lengthy. non blocker but i would consider breaking this up into seperate methods.